### ISSUE TITLE
Baking fails for tables with a numeric field name.

### ISSUE THREAD

**Issue by VARGA-Peter** [cakephp/bake#758]
For a table having a field name which is pure numeric [`1` for example] baking fails:

```
Error: [TypeError] Cake\Database\Schema\TableSchema::getColumn():
Argument cakephp/cakephp#1 ($name) must be of type string, int given, called in
.../vendor/cakephp/bake/src/Command/FixtureCommand.php on line 254 in
.../vendor/cakephp/cakephp/src/Database/Schema/TableSchema.php on line 351

Error: [TypeError] preg_match(): Argument cakephp/cakephp#2 ($subject) must be of type string, int given in
.../vendor/cakephp/bake/src/Command/ModelCommand.php on line 301
```

This table doesn't follow **ANY of** the CakePHP table naming conventions [it's a 3rd party product which tables are needed in the CakePHP project and read by manually created `SELECT` statements].

My opinion is that the baking should be stable as the table/field name can be called what ever a programmer wants using the backticks.

May be the CakePHP developers reconsider it and fix this _bagatelle_.

_My personal note_: I think it's unbelievable _stupid_ to use a number as a field name but the world isn't perfect.

**Comment by VARGA-Peter** [comment]
> Do you have identifier quoting enabled for your connection? Without that backticks will not be used.

Please advise. How should I enable the quoting for the `bake` script? In the project there is no problem.

**Comment by markstory** [comment]
https://book.cakephp.org/4/en/orm/database-basics.html#identifier-quoting

If the columns are working ok in the application code, there might just need to be some additional casts added to make column names into strings as they are coming back as integers.

**Comment by VARGA-Peter** [comment]
> https://book.cakephp.org/4/en/orm/database-basics.html#identifier-quoting
> 
> If the columns are working ok in the application code, there might just need to be some additional casts added to make column names into strings as they are coming back as integers.

I found the quoting documentation earlier but I don't understand how should I enable it for a connection in the baking script as it's per connection. Do I misunderstand something?

I wrote

_[it's a 3rd party product which tables are needed in the CakePHP project and read by manually created SELECT statements]_

so there is no need for quoting.

**Comment by markstory** [comment]
> I found the quoting documentation earlier but I don't understand how should I enable it for a connection in the baking script as it's per connection. Do I misunderstand something?

If identifier quoting isn't enabled for in your applications' configuration select statements can fail when column names use SQL keywords or other identifiers that could be confused with SQL primitives. For example `SELECT 1, 2, 3 FROM table_name` could be column names or literal values.

> so there is no need for quoting.

Ok. As you're aware bake isn't really intended to work well on schemas that don't follow the conventions. If you really need to use bake on these tables. We'll need some help in reproducing the problems, or you'll need to send a pull request with a fix.

**Comment by VARGA-Peter** [comment]
Create the below table and run for the database `bin/cake bake model all` and you see it fails.

```
CREATE TABLE `bar` (
  `1` int(11) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
COMMIT;
```

**Comment by othercorey** [comment]
What is expected out come of baking a table with column name `1`? We can't generate field names for it.

**Comment by VARGA-Peter** [comment]
> What is the expected outcome when baking a table with column name `1`? We can't generate field names for it.

I would expect a message from the baking script informing the programmer that column names like this cannot be processed and that this table has been ignored. For sure I am not expecting a PHP error and the whole project breaks and remains in an inconsistent state. Guys, I hope you understand.

I wrote that I cannot believe somebody is using column names like this, but the world isn't perfect.
I am convinced it's obvious that the bake script has to catch such stupid column names and must not exit with a PHP error.

**Comment by othercorey** [comment]
I'm not arguing it should break. I was just curious how much we want to support it.

**Comment by ADmad** [comment]
> I was just curious how much we want to support it.

I wouldn't put any effort into this except putting some checks and bailing out when batshit crazy column names are encountered.

**Comment by VARGA-Peter** [comment]
> > I was just curious how much we want to support it.
> 
> I wouldn't put any effort into this except putting some checks and bailing out when batshit crazy column names are encountered.

Yes, exactly. Ignore the whole table, show a message and continue with the next one.
